src/tools/idlpp/src/mbchar.c

```diff
@@ -381,11 +382,19 @@ const char *    set_encoding(
             mcpp_fputc( '\n', ERR);
         }
     }
+    /* XPG syntax: language[_territory[.codeset]][@modifier]    */
     len = strlcpy( norm, name, sizeof(norm));
-    if (len >= 5 && norm[ 5] == '.')
-        memmove( norm, norm + 5, (len - 5) + 1);
+    for (pos = 0; pos < len && norm[pos] != '@'; pos++)
```

Comment:
According the man pages, `strlcpy` always returns the `strlen(source)`, which means this loop will read beyond `norm` if `strcspn(name,"@")>=sizeof(norm)`.

Otherwise, fine with me, if a bit convoluted. (I'd have factored it out into a separate function and used `strchr`, and made `norm` a pointer into a array of `char`, because that'd save me from having to do the math on a `memmove` 😁)

Reply:
Good point! Thanks @eboasson